File: Sliding Window/1695. Maximum Erasure Value.py
class Solution(object):
    def maximumUniqueSubarray(self, nums):
        seen = set()
        ans, sum, i, j, n = 0, 0, 0, 0, len(nums)
        while j < n:
            sum += nums[j]
            while nums[j] in seen:
                sum -= nums[i]
                seen.remove(nums[i])
                i += 1
            seen.add(nums[j])
            ans = max(ans, sum)
            j += 1
        return ans

File: Sliding Window/1695. Maximum Erasure Value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) 
    {
        unordered_set<int> seen;
        int maxSum = 0, sum = 0;
        int i = 0, j = 0, n = nums.size();
        while(j < n)
        {
            sum += nums[j];
            while(seen.count(nums[j]))
            {
                sum -= nums[i];
                seen.erase(nums[i++]);
            }
            seen.insert(nums[j++]);
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};